addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -339,9 +339,16 @@ export class ProductScreen extends Component {
             }
         }
 
+        const customSearchField = this.pos.session._pos_product_sort_int_field;
+
         return this.searchWord !== ""
             ? filteredList
-            : filteredList.sort((a, b) => a.display_name.localeCompare(b.display_name));
+            : filteredList.sort((a, b) => {
+                  if (customSearchField) {
+                      return a[customSearchField] - b[customSearchField];
```

Comment:
What if custom search field is "name", should we use localcompare too ?

Reply:
@davidmonnom I limited the custom field to be only integer https://github.com/odoo/odoo/pull/192094/files#diff-62e5deb63b22181cdbbf872d106e3e528a72a80a8c7fbaa2cf5d2c6f7309ed29R277

Reply:
@davidmonnom Could you please review this pr?